src/main/java/com/emc/pravega/perf/PravegaStreamHandler.java

```diff
@@ -0,0 +1,159 @@
+            return;
+        }
+
+        System.out.println("The stream: " + stream + " will be manually scaling to " + segCount + " segments");
+
+        /*
+         * Note that the Upgrade stream API does not change the number of segments;
+         * but it indicates with new number of segments.
+         * after calling update stream , manual scaling is required
+         */
+        if (!streamManager.updateStream(scope, stream, streamconfig)) {
+            throw new TimeoutException("Could not able to update the stream: " + stream + " try with another stream Name");
+        }
+
+        final double keyRangeChunk = 1.0 / segCount;
+        final Map<Double, Double> keyRanges = IntStream.range(0, segCount)
+                                                       .boxed()
+                                                       .collect(
+                                                           Collectors
+                                                               .toMap(x -> x * keyRangeChunk,
+                                                                   x -> (x + 1) * keyRangeChunk));
+        final List<Long> segmentList = segments.getSegments()
+                                               .stream()
+                                               .map(Segment::getSegmentId)
+                                               .collect(Collectors.toList());
+
+        CompletableFuture<Boolean> scaleStatus = controller.scaleStream(new StreamImpl(scope, stream),
+            segmentList,
+            keyRanges,
+            bgexecutor).getFuture();
+
+        if (!scaleStatus.get(timeout, TimeUnit.SECONDS)) {
+            throw new TimeoutException("ERROR : Scale operation on stream " + stream + " did not complete");
+        }
+
+        System.out.println("Number of Segments after manual scale: " +
+            controller.getCurrentSegments(scope, stream)
+                      .get().getSegments().size());
+    }
+
+    void recreate() throws InterruptedException, ExecutionException, TimeoutException {
+        System.out.println("Sealing and Deleteing the stream : " + stream + " and then recreating the same");
+        CompletableFuture<Boolean> sealStatus = controller.sealStream(scope, stream);
+        if (!sealStatus.get(timeout, TimeUnit.SECONDS)) {
+            throw new TimeoutException("ERROR : Segment sealing operation on stream " + stream + " did not complete");
+        }
+
+        CompletableFuture<Boolean> status = controller.deleteStream(scope, stream);
+        if (!status.get(timeout, TimeUnit.SECONDS)) {
+            throw new TimeoutException("ERROR : stream: " + stream + " delete failed");
+        }
+
+        if (!streamManager.createStream(scope, stream, streamconfig)) {
+            throw new TimeoutException("ERROR : stream: " + stream + " recreation failed");
+        }
+    }
+
+    ReaderGroup createReaderGroup() throws URISyntaxException {
+        if (readerGroup != null) {
```

Comment:
This is not thread safe.

Reply:
Additionally I'm not sure why it needs to guard against it being called multiple times. I only see one call site, and it only is invoked once in the lifetime of the process. I think that readerGroup can simply be returned and not stored as a member variable.